views/js/testrunner.js

```diff
@@ -87,10 +85,48 @@ define(['jquery', 'lodash',  'i18n', 'helpers', 'iframeNotifier', 'serviceApi/Se
                     else {
                         this.testServiceApi.finish();
                     }
+                    if(data.previous){
+                        $('#previous').removeClass('hidden');
+                    }
+                    else{
+                        $('#previous').addClass('hidden');
+                    }
                 }
             });
 
         },
+
+        previousItem: function() {
+
+            iframeNotifier.parent('loading');
+            var that = this;
+
+            this.currentItemApi.kill(function(signal) {
+                that.doPrevious();
+                iframeNotifier.parent('unloading');
+            });
+        },
+
+        doPrevious: function() {
+                $.ajax({
+                    context: this,
+                    url: helpers._url('previous', 'TestRunner', 'taoTestLinear'),
+                    data: { serviceCallId: this.testServiceApi.getServiceCallId() },
+                    accepts: 'application/json',
+                    cache: false,
+                    type: 'POST',
+                    success: function(data, textStatus, jqXhr) {
+                        if (data.api !== null) {
+                            var itemServiceApi = eval(data.api);
```

Comment:
`eval` shouldn't be used. This is a huge security issue (see XSS), but if you can't change the server API, at least check that itemServiceApi is what you expect